### DISCUSSION TITLE
Is there any way to disable the upgrade message?

### DISCUSSION THREAD

**Discussion by williamstein** [fonsp/Pluto.jl#2364] [Questions about using Pluto] [answered]
Is there any way to disable this popup appearing?

<img width="818" alt="image" src="https://user-images.githubusercontent.com/1276278/201382859-82811d7e-f086-4885-8f17-1d7cb9fa44ab.png">

I operate a service https://cocalc.com that provides hosted Pluto.jl notebooks.  However, the installation is system-wide, so users do not have permissions to modify their Pluto installation.  So the above upgrade notice is just very confusing and frustrating to them.  

We do upgrade regularly, but our upgrade cycle isn't exactly the same as Pluto's. 

In general it is also a little concerning that Pluto.jl is perhaps "phoning home"  to check for upgrades (I don't know how the version check works, but it might be used to gather info about where users of Pluto are located).  Is there a way to "disable all telemetry"?

Thanks!

**Comment by williamstein** [comment]
Note that I'm even willing to modify the source code of Pluto.jl via a patch as part of my install process, if that's the only way.

**Reply by pankgeorg** [reply]
We use github to check the versions - we do not "phone home" in any traditional sense. We may be using a version of [goatcounter](https://www.goatcounter.com/) for some stuff. 

For the releases, see `frontend/common/NewUpdateMessage.js`. We use jsdeliver and github as third party javascript sources, even though we bundle the frontend code in releases (so if you're using `] add Pluto@release` you get a bundled frontend but if you use a branch you get an unbundled version that may use jsdeliver or other CDNs.

Note that, as a rule of thumb, we don't make many guarantees about avoiding using publicly available services, we have optimized for speed, not security/compliance! If that is of concern to you, it's best to take a look at our frontend source code.

For your specific scenario, I think the best way to go is to use a small build script: Checkout the branch/release, remove the file from the tree, change line 40 to `}).filter(x => false)` and trigger a build of the frontend. That can go in your Dockerfile or something similar, but completely avoiding something hacky (like a sed script) will not be very easy.

Hope that helps!

**Reply by williamstein** [reply]
> We use github to check the versions - we do not "phone home" in any traditional sense.

Cool.  **Thanks for confirming that, and I appreciate the clarity.**

Also many thanks to the source link!  I can see from browsing the code that the answer to my question is: "NO".

A follow up question to completely solve this thread is: "how can I modify the source code to disable the upgrade message"? and it seems like the answer is to change the function ` fetch_pluto_releases` at

https://github.com/fonsp/Pluto.jl/blob/main/frontend/common/NewUpdateMessage.js#L30

Would something like this work?
```jl
const fetch_pluto_releases = async () => { return []; }
```

Or maybe

```jl
const fetch_pluto_releases = async () => { error("disabled"); }
```

I want to change `fetch_pluto_releases` to avoid making an unnecessary network connection.

Thanks!!

**Reply by williamstein** [reply]
Thanks.  I see you updated your answer (or something changed) to answer the question I asked while responding.  

>  That can go in your Dockerfile or something similar, but completely avoiding something hacky (like a sed script) will not be very easy.

I'll likely just use a patch.

Would you be open to us submitting a PR to Pluto that changes the behavior of fetch_pluto_release depending on an environment variable?   That would probably be the best longterm solution from my point of view.  Other people running pluto on multiuser systems or cloud hosted environments who run into the same issue would then benefit.

**Comment by williamstein** [comment]
Another option might be to stat some files in the Pluto distribution and if the user running Julia doesn't have write permissions to them, show no message or a different message?

**Reply by pankgeorg** [reply]
We prefer automatically inferring stuff about the environment vs adding environment variables that users may start questioning whether they should use and when. Pluto is meant to be used for education and students, so we want to keep it as simple as possible. In that spirit, Pluto respects very small set of environment variables.

We do though try to detect stuff like WSL, so I think a stat-based solution would be within the project philosophy! Looking forward to see it!

**Comment by fonsp** [comment, marked as answer]
It's actually easier than this, you can run Pluto with:

```julia
Pluto.run(dismiss_update_notification=true)
```

PS I love your work on cocalc.com , really cool to see Pluto there! Let us know if we can do anything else to support it :)

**Reply by williamstein** [reply]
cool, thanks!
